src/middleware.ts

```diff
@@ -259,35 +262,55 @@ export const persist = <S extends State>(
   }
 
   // rehydrate initial state with existing stored state
-  ;(async () => {
+  ;(() => {
     const postRehydrationCallback = onRehydrateStorage?.(get()) || undefined
 
-    try {
-      const storageValue = await storage.getItem(name)
+    let storageValueUnknown = storage.getItem(name)
 
-      if (storageValue) {
-        const deserializedStorageValue = await deserialize(storageValue)
-
-        // if versions mismatch, run migration
-        if (deserializedStorageValue.version !== version) {
-          const migratedState = await migrate?.(
-            deserializedStorageValue.state,
-            deserializedStorageValue.version
-          )
+    if (storageValueUnknown instanceof Promise) {
+      storageValueUnknown
+        .then((storageValue) => storageValue && deserialize(storageValue))
```

Comment:
The original code doesn't seem to check `storageValue`.
```suggestion
        .then((storageValue) => deserialize(storageValue))
```

Reply:
dear @dai-shi, the original source code also checks the storage value before serializing.
```Javascript
const storageValue = await storage.getItem(name)
if (storageValue) {        
   const deserializedStorageValue = await deserialize(storageValue)
```

Reply:
oops, sorry. looks like i missed it.